blocks/v2-hero/v2-hero.js

```diff
@@ -9,10 +9,44 @@ import {
 } from '../../scripts/common.js';
 
 const variantClasses = ['centered', 'left', 'bottom', 'dark'];
+let intervalId = null;
+const blockName = 'v2-hero';
 
-export default async function decorate(block) {
-  const blockName = 'v2-hero';
+function updateCountdown(eventTime, block) {
+  const now = new Date();
+  const diff = eventTime - now;
+
+  // Check if the event time has passed
+  if (diff <= 0) {
+    block.querySelector(`.${blockName}__countdown-wrapper`).remove();
+    clearInterval(intervalId);
+    return;
+  }
+
+  // Calculate time left
+  const days = Math.floor(diff / (1000 * 60 * 60 * 24));
+  const hours = Math.floor((diff % (1000 * 60 * 60 * 24)) / (1000 * 60 * 60));
+  const minutes = Math.floor((diff % (1000 * 60 * 60)) / (1000 * 60));
+  const seconds = Math.floor((diff % (1000 * 60)) / 1000);
 
+  // Format labels
+  const dayLabel = days > 1 ? 'days' : 'day';
+  const hourLabel = hours > 1 ? 'hours' : 'hour';
+  const minuteLabel = minutes > 1 ? 'minutes' : 'minute';
+  const secondLabel = seconds > 1 ? 'seconds' : 'second';
+
+  block.querySelector('#days').textContent = days.toString().padStart(2, '0');
```

Comment:
Can you change to this to below lines , it would reduce query of elements.
```
const dayEle = block.querySelector('#days');
  dayEle.textContent = days.toString().padStart(2, '0');
  dayEle.nextElementSibling.textContent = dayLabel;
```